doc/example_code/how_to_examples/joystick_controller.rst

```diff
@@ -0,0 +1,169 @@
+* The x-axis numbers will be negative if the stick goes left, positive for right.
+* The y-axis numbers will be opposite of what you might expect. Negative for up, positive for down.
+
+.. list-table:: joystick controller
+   :widths: 25 25 50
+   :header-rows: 1
+   
+   * - Left & Right
+     - Centered
+     - Up & Down
+   * -   (-1, 1)
+     -    0
+     -   (-1, 1)
+
+
+The Joystick Movements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. image:: joystickController.png
+    :width: 400px
+    :align: center
+    :alt: Screenshot of controller movements
+
+     
+Deadzone
+========
+A centered joystick might have a value not at 0, but at 0.0001 or some small number. This will make for a small “drift” on a person’s character. We often counteract this by having a “dead zone” where if the number is below a certain value, we just assume it is zero to eliminate the drift.
+
+How we take care of the dead zone:
+
+After 
+
+.. code-block:: console
+      import arcade
+
+add the following line at the top of the code to define a constant :code:`DEAD_ZONE` :
+
+.. code-block:: console
+
+    DEAD_ZONE = 0.02
+
+
+and adding the following code to the :code:`update`:
+
+.. code-block:: python
+
+    def update(self, delta_time):
+
+            # Update the position according to the game controller
+            if self.joystick:
+
+                # Set a "dead zone" to prevent drive from a centered joystick
+                if abs(self.joystick.x) < DEAD_ZONE:
+                    self.object.change_x = 0
+                else:
+                    self.object.change_x = self.joystick.x * MOVEMENT_SPEED
+
+                # Set a "dead zone" to prevent drive from a centered joystick
+                if abs(self.joystick.y) < DEAD_ZONE:
+                    self.object.change_y = 0
+                else:
+                    self.object.change_y = -self.joystick.y * MOVEMENT_SPEED
+
+
+Different Types Of Controllers
+==============================             
+Some types of controllers:
+
+1. Joystick hats: the directional pads on game controller. It allows you to move in eight directions (up, down, left, right and the diagonals).
+
+2. Xbox controller: This is a popular gamepad used with Xbox consoles and Windows PCs. It has two thumbsticks, a directional pad, and various buttons.
+
+3. PlayStation controller: This is the gamepad used with PlayStation consoles. It has two thumbsticks, a directional pad, and various buttons.
+
+4. Nintendo Switch controller: This is the gamepad used with Nintendo Switch consoles. It has two detachable Joy-Con controllers, each with a joystick, buttons, and motion sensors.
+
+5. Logitech G29 racing wheel: This is a steering wheel controller designed for racing games. It has a wheel, pedals, and various buttons.
+
+6. Thrustmaster T.Flight Hotas X flight stick: This is a flight stick controller designed for flight simulation games. It has a joystick, throttle, and various buttons.
```

Comment:
The racing wheel and flight stick should probably be removed from this list. They're unlikely to work well with the controller API, they would be better suited for the legacy joystick API in Pyglet, which isn't covered by this documentation. The controller API is really targeted at a "traditional" hand held controller.